The new line at the end need to be added here to mitigated differences
between Solaris and GNU grep commands. Where GNU grep will add new line
at the end when used on file without it.

--- libxslt-1.1.37/tests/docbook/Makefile.am
+++ libxslt-1.1.37/tests/docbook/Makefile.am
@@ -68,6 +68,7 @@ out=result/xhtml/`basename $$i .xml`.tst
 	  xhtml=result/xhtml/`basename $$i .xml`.xhtml; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc --nonet \
 		$(srcdir)/xhtml/docbook.xsl $$i > $$out ; \
+	  echo '' >> $$out ; \
 	  if [ -f $(srcdir)/$$xhtml ] ; then \
 	      grep -v id < $(srcdir)/$$xhtml > $$xhtml.noid ; \
 	      grep -v id < $$out > $$out.noid ; \
